lib/matplotlib/backends/backend_wx.py

```diff
@@ -810,12 +789,41 @@ def _onPaint(self, evt):
         """
 
         DEBUG_MSG("_onPaint()", 1, self)
-        drawDC = wx.PaintDC(self)
         if not self._isDrawn:
-            self.draw(drawDC=drawDC)
+            self.draw()
+
+        # the bitmap can not be in use by another DC
+        img = self.bitmap.ConvertToImage()
+        bmp = img.ConvertToBitmap()
+        dc = wx.BufferedPaintDC(self, bmp)
+
+        if not self._rubberband:
+            return
+
+        # draw rubberband / selection using an overlay
+        if self._overlay is None:
+            self._overlay = wx.Overlay()
+        odc = wx.DCOverlay(self._overlay, dc)
+
+        if 'wxMac' not in wx.PlatformInfo:
+            # draw a box with border and 50% transparency
+            dc = wx.GCDC(dc)
+            # Set the pen, for the box's border
+            bc = wx.BLUE
+            dc.SetPen(wx.Pen(colour=bc, width=1, style=wx.PENSTYLE_SOLID))
+            # Create a brush (for the box's interior) with the same colour,
+            # but 50% transparency.
+            bc = wx.Colour(bc.red, bc.green, bc.blue, 0x80)
+            dc.SetBrush(wx.Brush(bc))
         else:
-            self.gui_repaint(drawDC=drawDC)
-        drawDC.Destroy()
+            # draw without transparency which is buggy on Retina displays
+            dc.SetPen(wx.Pen(wx.BLUE, 1, wx.SOLID))
```

Comment:
The blue seems a bit deeper than in the original version.  Also none of the other backends (well, dunno for macos) actually color the rectangle, so perhaps we may just leave it as transparent).